make-util.js

```diff
@@ -1895,65 +1895,49 @@ exports.generateTasks = generateTasks;
  * @returns {Function} - wrapped buildTask function which compares diff between source and generated tasks
  * and copy files from generated to source if needed
  */
-function syncGeneratedFilesWrapper(originalFunction, genTaskPath, callGenTaskDuringBuild = false) {
-    const allowedFilesToCopy = ["package.json", "package-lock.json", "npm-shrinkwrap.json"];
-    const genTaskBasePath = path.basename(genTaskPath);
+function syncGeneratedFilesWrapper(originalFunction, basicGenTaskPath, callGenTaskDuringBuild = false) {
+    const runtimeChangedFiles = ["package.json", "package-lock.json", "npm-shrinkwrap.json"];
 
     if (!originalFunction || originalFunction instanceof Function === false) throw Error('originalFunction is not defined');
     // If the task is building on the ci, we don't want to sync files
     if (callGenTaskDuringBuild === false) return originalFunction;
 
-    console.log(
-        "Syncing generated files with source task...\n" +
-        "----------------------------------------------\n" +
-        "Getting list of uncommitted changes");
-    const initialDiffOutput = run(`git -C "${repoPath}" diff --name-only`)
-    console.log(
-        "uncommitted changes:\n" + 
-        `${initialDiffOutput}\n` +
-        "----------------------------------------------");
-
     return function(taskName, ...args) {
-        // git diff --name-only return "/"" as separator
         originalFunction.apply(this, [taskName, ...args]);
 
-        const genTaskPath = `${genTaskBasePath}/${taskName}/`;
+        const genTaskPath = path.join(basicGenTaskPath, taskName);
+
         // if it's not a generated task, we don't need to sync files
         if (!fs.existsSync(genTaskPath)) return;
+        const [baseTaskName, config] = taskName.split("_");
 
-        const changedPaths = [];
-        const afterTaskBuildDiff = run(`git diff --name-only`);
-
-        afterTaskBuildDiff.split("\n").forEach((line) => {
-            // We skip files that are not in the generated task folder or are already in the initial diff   
-            if (!line.includes(genTaskPath) || initialDiffOutput.includes(line)) return;
-            changedPaths.push(line);
-        });
-
-        if (changedPaths.length === 0) return;
-
-        console.log(
-            "The following generated files where updated in source task after build and added to staged, please commit them to the repo:\n" +
-            `${changedPaths}\n` +
-            "----------------------------------------------\n" +
-            "The following files were copied to source task:");
-
-        changedPaths.forEach((filePath) => {
-            const fileName = path.basename(filePath);
-            if (allowedFilesToCopy.indexOf(fileName) === -1) return;
-            const [baseTaskName, config] = taskName.split("_");
+        runtimeChangedFiles.forEach((fileName) => {
+            const filePath = path.join(genTaskPath, fileName);
```

Comment:
>filePath = path.join(genTaskPath, fileN [](http://example.com/codeflow?start=18&length=39)

I believe this misses files in subdirectories e.g. Test\package-lock.json

Reply:
Fixed, thanks